### PR TITLE
[IMP] mail, im_livechat: kwargs in mail_message_post params

### PR DESCRIPTION
**Pull request by maki-odoo**
This commit adds `kwargs` to `mail_message_post()` params in order to split the data related to post a message and others.

### CODE REVIEW

File: addons/mail/controllers/thread.py [hunk 1]
```diff
@@ -72,14 +72,16 @@ def _get_allowed_message_post_params(self):
 
     @http.route("/mail/message/post", methods=["POST"], type="json", auth="public")
     @add_guest_to_context
-    def mail_message_post(self, thread_model, thread_id, post_data, context=None):
+    def mail_message_post(self, thread_model, thread_id, post_data, extra_data=None, context=None):
```
Comment (seb-odoo) [inline addons/mail/controllers/thread.py:75]: Discussion about using `**kwargs` or listing the params instead of having `extra_data`

**Review by seb-odoo — COMMENTED** [review]
Isn't there a mock server method to be updated somewhere too?

I'm surprised this is green actually, I guess we never "qunit/hoot" test these features? Would it be possible to add them?

**Comment by maki-odoo** [comment]
> Isn't there a mock server method to be updated somewhere too?
> 
> I'm surprised this is green actually, I guess we never "qunit/hoot" test these features? Would it be possible to add them?

No, We don't have any "qunit/hoot" test for them. I tri

**Comment by maki-odoo** [comment]
> Isn't there a mock server method to be updated somewhere too?
> 
> I'm surprised this is green actually, I guess we never "qunit/hoot" test these features? Would it be possible to add them?

Test added for posting canned response

File: addons/mail/static/tests/composer/composer.test.js [hunk 2]
```diff
@@ -1035,3 +1036,25 @@ test("Canned response can be deleted from the bus", async () => {
...
+    await start();
+    await openDiscuss(channelId);
+    await insertText(".o-mail-Composer-input", ":");
+    await click(".o-mail-NavigableList-item", { text: "testTest a canned response?" });
+    await contains(".o-mail-Composer-input", { value: "Test a canned response? " });
+    expect(cannedResponse.last_used).toBe(false);
+    await click(".o-mail-Composer-send:enabled");
+    await contains(".o-mail-Message");
+    cannedResponse = pyEnv["mail.canned.response"].search_read([["id", "=", cannedResponseId]])[0];
+    expect(cannedResponse.last_used).toBe(serializeDateTime(today()));
```
Comment (seb-odoo) [inline addons/mail/static/tests/composer/composer.test.js:1059]: I don't know if using today() is reliable when running the test around midnight, maybe just checking there is a value is enough, the specific value is not so important.

File: addons/mail/static/tests/mock_server/mail_mock_server.js [hunk 3]
```diff
@@ -534,8 +534,39 @@ export async function mail_message_post(request) {
...
+        thread_id,
+        thread_model,
+        partner_emails,
+        partner_additional_values,
+        canned_response_ids,
+    } = await parseRequestParams(request);
+    if (canned_response_ids) {
+        for (const cannedResponseId of canned_response_ids) {
+            this.env["mail.canned.response"].write([cannedResponseId], {
+                last_used: serializeDateTime(today()),
```
Comment (seb-odoo) [inline addons/mail/static/tests/mock_server/mail_mock_server.js:552]: Why `today` instead of `now`?

**Review by seb-odoo — COMMENTED** [review]
PR will need to be rebased, but otherwise the principle looks good.

**Review by seb-odoo — APPROVED** [review]
@robodoo r+

**Pull request closed without merging**